### PR TITLE
Nicer tests reporter and interface

### PR DESCRIPTION
**Pull request by lieuwex**

### CODE REVIEW

File: test/test.c [hunk 1]
```diff
@@ -51,58 +66,86 @@
...
+		CHECKJSONERR("{\"obj\":{\"1\":2.\"3\":4},\"x\":y}");
+		CHECKJSON("{ \"kaas\": [ \"is\", \"lekker\" ] }");
+
+		CHECKBIDIREQ("{\"a\":\"\\u003c\t\n\fkaas\\\"\",\"iets\":[]}",
+		             "{\"a\":\"<\\t\\n\\fkaas\\\"\",\"iets\":[]}");
+	});
+
+	bool successful = passedTotal == ranTotal;
+	printf(color("%d failed out of %d\n", successful ? "green" : "red"), ranTotal - passedTotal, ranTotal);
+	return successful ? 0 : 1;
```
Comment (lieuwex) [inline test/test.c]: @tomsmeding It also returns a correct exit code, so we can also use travis-ci for this repo.

**Comment by lieuwex** [comment]
I must also say that this reporter is designed for 'interactive' use, not to be parsed whatsoever.

**Comment by tomsmeding** [comment]
Only for interactove use is fine. I think travis is a bit overkill, since it's not exactly hard to run the tests, and in fact they already run on every `make`. But +1 to this new format, haven't tested it yet though.

Your `color` function allocates a string with `asprintf`, but in no occasion does that string get freed again I think. But then again, my json check macro's didn't free anything either, so we're both guilty :p This needs to be fixed though, but I can do it if you wish.

**Comment by lieuwex** [comment]
> Your color function allocates a string with asprintf, but in no occasion does that string get freed again I think. 

Yeah I know I don't clear free the strings, but I didn't really know a nice way of doing that. And strings don't really take that much memory, and everything will be cleared by the OS when the program exists, right? So I don't think it's really a big problem.

> This needs to be fixed though, but I can do it if you wish.

That would be great :)

> I think travis is a bit overkill, since it's not exactly hard to run the tests, and in fact they already run on every `make`.

Haha it's not that much work to add, I can do it if you want, and it looks nice in the README ;)

**Comment by tomsmeding** [comment]
I'll merge this and then fix the string free'ing on master.

**Comment by lieuwex** [comment]
🎉

**Pull request merged**